givitsite/accounts/views.py

```diff
@@ -2,42 +2,25 @@
 from django.contrib.auth import authenticate
 from django.contrib.auth.models import auth, User
 from django.contrib import messages
-from accounts.forms import RegistrationForm
-
-# Create your views here.
+from .forms import RegistrationForm, PersonInfoForm
 
 def register(request):
-
     if request.method == 'POST':
-        form = RegistrationForm(request.POST)
-        if form.is_valid():
-            form.save()
-        return redirect('/')
-    
-    else:
-        form = RegistrationForm()
-        args = {'form': form}
-    return render(request, 'register.html', args)
+        user_form = RegistrationForm(request.POST)
+        profile_form = PersonInfoForm(request.POST)
+        if user_form.is_valid() and profile_form.is_valid():
+            user = user_form.save()
+            profile = profile_form.save(commit=False)
+            profile.user = user
 
-    #     if password1 == password2:
-    #         if User.objects.filter(username=username).exists():
-    #             messages.info(request,'Username Taken')
-    #             return redirect('register')
-    #         elif User.objects.filter(email=email).exists():
-    #             messages.info(request,'email Taken')
-    #             return redirect('register')
-    #         else:
-    #             user = User.objects.create_user(username = username, email = email, password = password1, first_name = first_name, last_name = last_name, address_street = address_street, date_of_birth = date_of_birth, address_home_number = address_home_number, address_city = address_city, phone_number = phone_number, field_of_study = field_of_study, degree = degree, current_year = current_year, scholarship = scholarship, type_of_friendship = type_of_friendship)
-    #             user.save();
-    #             return redirect('home.html')
-    #     else:
-    #         messages.info(request,'password not match')
-    #         return redirect('register')
-    #     return redirect('/')
-
-    # else:
-    #     return render(request, 'register.html')
+            profile.save()
 
+            return redirect('/')
+    else:        
```

Comment:
does this mean that if the form is invalid you return empty forms to the user? It might be nicer to keep the data they already typed and just show a warning message (I wouldn't be surprised if Django does that automatically...)

Reply:
You right in Use field django does it automatically all I need to do is display it on the screen. Other fields I need to do it manually. I will open a ticket on it.